src/main/java/com/google/summit/SummitAST.kt

```diff
@@ -148,7 +148,7 @@ object SummitAST {
    *
    * After execution, the input cursor of the [charStream] is reset to index 0.
    */
-  private fun determineCompilationType(charStream: CharStream): CompilationType {
+  fun determineCompilationType(charStream: CharStream): CompilationType {
```

Comment:
I had to make `determineCompilationType` public to access it from the tests. Is there a better way to do this?

Reply:
If you think this should be part of the public API (for example, if it might conceivably be useful to callers), then this is an ok result.

Otherwise, I think the answer boils down to: http://go/unit-testing-practices#public-apis

Reply:
I was thinking about making `determineCompilationType` `internal` instead, and using [`associates`](https://bazelbuild.github.io/rules_kotlin/kotlin.html#kt_jvm_library-associates) to make it visible from the unit test. This seems to match the intended behavior of `internal` (e.g. see Gradle in [Modules](https://kotlinlang.org/docs/visibility-modifiers.html#modules)) but goes against the advice in that link.

Ideally, `determineCompilationType` wouldn't be exposed as the compilation type detection isn't intended to be done by the user directly, but the solutions proposed in the link also don't seem preferable. If we were to try to test this operation through the public interface (`parseAndTranslate`), for example, we wouldn't get useful results for syntactically invalid input strings (parse error) that could still be a valid input for `determineCompilationType`. A separate class seems like too much for this relatively small operation.

Reply:
> If we were to try to test this operation through the public interface (parseAndTranslate), for example, we wouldn't get useful results for syntactically invalid input strings (parse error) that could still be a valid input for determineCompilationType.

I understand http://go/unit-testing-practices#public-apis to be arguing that you _shouldn't_ test this internal detail (of what is invalid and valid for determineCompilationType), if it can't be observed from the public caller.